### ISSUE TITLE
[LSP]

### ISSUE THREAD

**Issue by SolaWing** [sorbet/sorbet#2496]
#### Input

`srb tc --lsp`
root directory contains a rbi folder

#### Observed output

error msg:  Sorbet's language server requires a single input directory.

use `srb tc --lsp --no-config --verbose`, seems rbi folder are implicit add to input directory

#### Expected behavior

can launch lsp

**Comment by connorshea** [comment]
Assuming you mean that gems with an `rbi/` directory cause the LSP to break, I think this is fixed by #2730?

**Comment by dogweather** [comment]
I'm seeing this too.

**Comment by feliperaul** [comment]
Here as well

**Comment by RB-Lab** [comment]
I'm trying to use it with this extension for VSCode https://github.com/danhuynhdev/sorbet-lsp, and still set this error. Sobet version in bundle is `sorbet (0.5.6307)`.

**Comment by kbacha** [comment]
I'm seeing this issue as well, both in the gem that has the `rbi/` directory and a gem that pulls it in as a dependency.

**Comment by Prajjwal** [comment]
Bump. Seeing the same error while trying to get it working with Vim + `ale`.

**Comment by Fire-Dragon-DoL** [comment]
I'm seeing this problem too. I have a gem with `rbi` directory (needed to make typings work), but with --lsp, it fails.

I must point out that being able to add multiple directories to the configuration is very important, I think the `--lsp` option should keep supporting multiple directories. Is this possible?
Does `--ignore` support wildcards?

**Comment by arubis** [comment]
With 7 confirmations (and add mine as 8th), I think it'd be appropriate to remove the `Unconfirmed` tag so this can get some love.

**Comment by mcraeklaura** [comment]
You can add mine as a 9th as well :)

**Comment by crokobit** [comment]
10th here :)

**Comment by arpitsr** [comment]
11th here :). Please can you help here.

Sorbet's language server requires a single input directory. However, 0 are configured: []

**Comment by styrmis** [comment]
Experiencing the same issue—this was working last week but reverting to an earlier version of Sorbet does not resolve the issue.

```
[T1266592][2022-06-21T09:59:09.801717] Debug logging enabled
[T1266592][2022-06-21T09:59:09.801778] Running sorbet version 0.5.10037 git 5395cb663a1fc8bf4ba7a129f4138e3596530d54 debug_symbols=true clean=0 with arguments: sorbet --lsp --no-config --verbose
[T1266592][2022-06-21T09:59:09.822809] Starting sorbet version 0.5.10037 git 5395cb663a1fc8bf4ba7a129f4138e3596530d54 debug_symbols=true clean=0 in LSP server mode. Talk ‘\r\n’-separated JSON-RPC to me. More details at https://microsoft.github.io/language-server-protocol/specification.If you're developing an LSP extension to some editor, make sure to run sorbet with `-v` flag,it will enable outputing the LSP session to stderr(`Write: ` and `Read: ` log lines)
[T1266592][2022-06-21T09:59:09.822842] Sorbet's language server requires a single input directory. However, 0 are configured: []
[T1266592][2022-06-21T09:59:09.822869] Write: {"jsonrpc":"2.0","method":"window/showMessage","params":{"type":1,"message":"Sorbet's language server requires a single input directory. However, 0 are configured: []"}}

Content-Length: 169

{"jsonrpc":"2.0","method":"window/showMessage","params":{"type":1,"message":"Sorbet's language server requires a single input directory. However, 0 are configured: []"}}
```

**Comment by paracycle** [comment]
@crokobit @arpitsr @styrmis You reports are not related to this issue at all. On the surface, the errors look similar to the original reported issue, but your problem is completely different.

Notice that the error you are getting says that 0 folders are configured, which means that there is a Sorbet configuration related issue. The original issue is about having more than 1 folder configured, and is a different and known limitation of Sorbet LSP mode.

@froydnj You should consider (and maybe move) the last 3 reports as a different issue.

**Comment by styrmis** [comment]
@paracycle You're right—my issue was resolved by a configuration change on my side (for the Neovim LSP) to correctly determine the root directory when working within a Rails engine.

Happy to delete my earlier comment (and this one!) to clean up the issue if you would prefer, as my issue is not a Sorbet one but one (potentially) for [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig).

**Comment by styrmis** [comment]
> @styrmis I suspect that, if you removed that flag, you wouldn't even need to change your NeoVim config.

Unfortunately this does not appear to be the case—the change as described is necessary (though there may be other ways to resolve the same issue).

The command run above was just following the form of the previous reports, which was misleading. The command being run by `nvim-lspconfig` was in fact just `srb tc --lsp`.

Apologies again for derailing this issue, I have confirmed that no gems with rbi directories have been added to this particular project, so my issue is indeed completely unrelated.

**Comment by dylanirlbeck** [comment]
Is there a way to configure Sorbet to accept more than one input directory? If not, how easy would it be to add support for multiple directories?

**Comment by Nakilon** [comment]
So how do I fix this? Sublime Text 4, ruby 2.7.2

**Comment by paulodiovani** [comment]
> @paracycle You're right—my issue was resolved by a configuration change on my side (for the Neovim LSP) to correctly determine the root directory when working within a Rails engine.
> 
> Happy to delete my earlier comment (and this one!) to clean up the issue if you would prefer, as my issue is not a Sorbet one but one (potentially) for [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig).

Hello,
can you share your config that solved the issue?

**Comment by ICHx** [comment]
I am configuring a multiroot workspace in vscode.  having the same error.  Adding `no-config` fixed my problem.

Here is my config
```
{
	"folders": [
		{
			"path": "../proj"
		},
		{
			"path": ".."
		},
	],
	"settings": {
		"sorbet.enabled": false,
		"sorbet.lspConfigs": [
			{
				"id": "stable2",
				"name": "Sorbet",
				"description": "Stable Sorbet Ruby IDE features",
				"command": [
					"bundle",
					"exec",
					"srb",
					"typecheck",
					"--lsp",
					"--disable-watchman",
					"--no-config"
				]
			},
		]
	}
}
```

**Comment by phdavis1027** [comment]
I'm getting this under 0.5.12115. Is there a recommended workaround for this? Also, I assume this means that LSP functionality like GotoDefinition doesn't work for symbols originating in external gems?

**Comment by Hedwig7s** [comment]
Also have this issue
This is an absolute deal breaker and frankly should be prioritized
